dlt/destinations/dataset/dataset.py

```diff
@@ -206,29 +202,32 @@ def __getattr__(self, table_name: str) -> ReadableIbisRelation:
 
     def __enter__(self) -> Self:
         """Context manager used to open and close sql client and internal connection"""
-        assert not self._sql_client, "context manager can't be used when sql client is initialized"
-        if not self._sql_client:
-            self._ensure_client_and_schema()
-            # return sql_client wrapped so it will not call close on __exit__ as dataset is managing connections
-            # use internal class
-
-            NoCloseClient = type(
-                "NoCloseClient",
-                (self._sql_client.__class__,),
-                {
-                    "__exit__": (
-                        lambda self, exc_type, exc_val, exc_tb: None
-                    )  # No-operation: do not close the connection
-                },
-            )
+        assert (
+            not self._opened_sql_client
+        ), "context manager can't be used when sql client is initialized"
+        # return sql_client wrapped so it will not call close on __exit__ as dataset is managing connections
+        # use internal class
+
+        # create a new sql client
+        self._opened_sql_client = self._get_sql_client(self.schema)
+
+        NoCloseClient = type(
```

Comment:
does this create a new object or modify the existing one?

Reply:
that modifies existing instance